src/lib/services/workflow-service.ts

```diff
@@ -293,30 +304,52 @@ export async function signalWorkflow({
   runId,
   signalName,
   signalInput,
+  settings,
+  accessToken,
 }: SignalWorkflowOptions) {
   const route = routeForApi('workflow.signal', {
     namespace,
     workflowId,
     runId,
   });
 
+  let payloads = null;
+
+  if (signalInput) {
+    if (settings?.codec?.endpoint) {
+      const awaitData = await convertPayloadsWithCodec({
+        payloads: { payloads: [JSON.parse(signalInput)] },
+        namespace,
+        settings,
+        accessToken,
+        encode: true,
```

Comment:
Should `encode` always be true here?

Reply:
Yeah should always be true if settings?.codec?.endpoint is set.